.github/workflows/run-security-scans.yml

```diff
@@ -189,26 +205,68 @@ jobs:
-            exit 1;
-          elif [[ $severity_check == 'true' && $EVENT_NAME == 'workflow_dispatch' ]]
+            if [[ $EVENT_NAME == 'pull_request' ]]
+            then
+              echo "Error: High or critical vulnerabilities detected on pull-request. Go to the Code Scanning section of the Github Security-tab and search for 'is:open pr:$pr_number' to review these vulnerabilities.";
+              exit 1;
+            elif [[ ($BRANCH == 'main' || $BRANCH == 'master')]]
+            then
+              echo "Error: High or critical vulnerabilities detected on main/master branch. Go to the Code Scanning section of the Github Security-tab to review these vulnerabilities.";
+              exit 1;
+            elif [[ $EVENT_NAME == 'workflow_dispatch' ]]
+            then
+              echo "Error: High or critical vulnerabilities detected on workflow-dispatch. Go to the Code Scanning section of the Github Security-tab and review vulnerabilities related to your workflow dispatch run.";
+              exit 1;
+            fi
+
+          elif [[ $ALLOW_SEVERITY_LEVEL == 'high' ]]
           then
-            echo "Error: High or critical vulnerabilities detected on workflow-dispatch. Go to the Code Scanning section of the Github Security-tab and review vulnerabilities related to your workflow dispatch run.";
-            exit 1;
-          else
-            echo "Success! No high or critical code scanning alerts."
+            if [[ ${{ steps.severity_check_outputs.outputs.severity_check_critical }} == 'true' ]]
+            then
+              echo "Only high vulnerabilities detected! Allowing due to input ALLOW_SEVERITY_LEVEL being set to high.";
+              exit 0;
+            fi
+
+            if [[ $EVENT_NAME == 'pull_request' ]]
+            then
+              echo "Error: Critical vulnerabilities detected on pull-request. Go to the Code Scanning section of the Github Security-tab and search for 'is:open pr:$pr_number' to review these vulnerabilities.";
+              exit 1;
+            elif [[ ($BRANCH == 'main' || $BRANCH == 'master')]]
+            then
+              echo "Error: Critical vulnerabilities detected on main/master branch. Go to the Code Scanning section of the Github Security-tab to review these vulnerabilities.";
+              exit 1;
+            elif [[ $EVENT_NAME == 'workflow_dispatch' ]]
+            then
+              echo "Error: Critical vulnerabilities detected on workflow-dispatch. Go to the Code Scanning section of the Github Security-tab and review vulnerabilities related to your workflow dispatch run.";
+              exit 1;
+            fi
+
+          elif [[ $ALLOW_SEVERITY_LEVEL == 'critical' ]]
+          then
+            echo "High or critical vulnerabilities detected! Allowing due to input ALLOW_SEVERITY_LEVEL being set to critical.";
+            exit 0;
           fi
```

Comment:
En bør implementere sikkert pattern for koding her og velge en sikker fail-safe default for denne if-else grenen. 
Dvs liste unntakene først og så legge det strengeste nivået til siste else-gren (som default).  

Jeg testet følgende kommando lokalt:
```
> severity_check=$(echo "test" | jq '[.[] | select(.state == "open" and (.rule.security_severity_level == "high" or .rule.security_severity_level == "critical"))] | any')
parse error: Invalid literal at line 2, column 0
> echo severity_check=$severity_check
echo severity_check=
>
```

if-else grenen i workflow vil da falle igjennom (spesielt for tilfellet allow_severity_level==`critical`).
Den tidligere versjonen hadde også denne svakheten der workflowikke ble avbrutt ved syntaxfeil.

Reply:
Vi sjekker jo etter severity_level syntaksen allerede i et eget steg:
<img width="1160" alt="image" src="https://user-images.githubusercontent.com/61270418/213391924-aef20337-34ce-4180-820d-b57117118da7.png">

Ettersom disse ikke er "koblet" sammen, så gir det dog mening å ha en failsafe, og da foreslår jeg i så fall at den failsafen er at alt kræsjer.